Update trajopt planner handling of fixed start and end states for collision

Levi-Armstrong opened the pull request:
This add ability to handle fixed states throughout the provided trajectory for collision. It also simplifies the trajopt configs to reduce code duplication. 

Update:
Noticed that there is an intermittent bug in OMPL hybridization which will return a plan that starts at the end and finishes at the end. Need to create issue on OMPL. 

Updated tesseract to support bullet being built using double precision.

Update rviz link_widget to support tesseract octree sub shape type SPHERE_INSIDE and SPHERE_OUTSIDE visualization.

mpowelson commented on tesseract/tesseract_planning/tesseract_motion_planners/include/tesseract_motion_planners/trajopt/config/trajopt_planner_default_config.h:
```diff
@@ -135,6 +135,19 @@ struct TrajOptPlannerDefaultConfig : public TrajOptPlannerConfig
+  bool checkUserInput() const;
+  bool addBasicInfo(trajopt::ProblemConstructionInfo& pci) const;
+  bool addInitTrajectory(trajopt::ProblemConstructionInfo& pci) const;
+  void addWaypoints(trajopt::ProblemConstructionInfo& pci, std::vector<int>& fixed_steps) const;
```
`void addWaypoints(trajopt::ProblemConstructionInfo& pci, std::vector<int>& fixed_steps = std::vector<int>())` ?

Levi-Armstrong replied:
I assume you are referring the fixed_steps parameter. It should be removed.

Levi-Armstrong replied:
Sorry I was incorrect. This is required. The fixed_steps are defined by the type of waypoints. If provided a waypoint of type JointWaypoint if would be set as a fixed_step all other types at the moment are considered not fixed steps.

mpowelson commented on tesseract/tesseract_planning/tesseract_motion_planners/src/trajopt/config/trajopt_planner_default_config.cpp:
```diff
@@ -142,143 +206,185 @@ std::shared_ptr<trajopt::ProblemConstructionInfo> TrajOptPlannerDefaultConfig::g
-      target_waypoints.back()->getType() == WaypointType::JOINT_TOLERANCED_WAYPOINT)
+}
+
+void TrajOptPlannerDefaultConfig::addConfiguration(trajopt::ProblemConstructionInfo& pci,
```
Perhaps we should make this addConfigurationCost or addKinematicConfiguration. addConfiguration sounds like you are adding a "config".

Levi-Armstrong replied:
Sure.

Levi-Armstrong replied:
done

Levi-Armstrong commented:
@johnwason Do you mind taking a look at the changes to the python in latest commit (Upate tesseract_python)?

Levi-Armstrong commented:
@johnwason I also found out why SWIG was not being found. The first issue was that FindSWIG looked for swig3.0 then swig2.0 then swig which is why it would find swig3.0 before the new version because it was named swig. I updated the install to rename swig executable to swig4.0 and added a new FindSWIG that looks for swig4.0 the others. With these changes it is no longer required to pass the compiler flag.

Levi-Armstrong commented:
@mpowelson Good to merge?

mpowelson commented:
@Levi-Armstrong Sure.

Levi-Armstrong commented:
I will remove the ci commit and merge.

johnwason commented:
@Levi-Armstrong the SWIG updates look fine

Levi-Armstrong commented:
@johnwason Thanks for taking a look.

hzyjerry commented:
I was able to build successfully on ubuntu 16.04. Thanks for the effort.

The key step was updating from swig3.0. To do that I downloaded latest swig4.0 from [this tutorial](https://github.com/swig/swig/wiki/Getting-Started). To install it, I also downloaded latest pcre based on [this issue](https://github.com/swig/swig/issues/1175)
